test/unit/Reflection/ReflectionObjectTest.php

```diff
@@ -106,6 +106,19 @@ public function testGetRuntimePropertiesWithFilter() : void
         self::assertEmpty($classInfo->getProperties(\ReflectionProperty::IS_PRIVATE));
     }
 
+    public function testGetRuntimeImmediatePropertiesWithFilter() : void
+    {
+        $foo = new \StdClass();
+        $foo->bar = 'huzzah';
+
+        $classInfo = ReflectionObject::createFromInstance($foo);
+
+        self::assertEmpty($classInfo->getImmediateProperties(\ReflectionProperty::IS_STATIC));
+        self::assertCount(1, $classInfo->getImmediateProperties(\ReflectionProperty::IS_PUBLIC));
```

Comment:
sorry @Ocramius but your comment is not clear to me...

In normal reflection, the method `getImmediateProperties` does not exist, and using `getProperties` the test would pass.

What do you mean by `runtime access will fail`?